### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by AndrMar1939**
[DEMO LINK](https://andrmar1939.github.io/react_todo-app-with-api/)

### CODE REVIEW

File: src/types/actionTypes.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+} | {
+  type: 'ALL_ACTIVE',
+  payload: null,
+} | {
+  type: 'IS_SPINNING',
+  payload: number,
+} | {
+  type: 'REMOVE_SPINNING',
+  payload: number,
+};
```
Comment (BudnikOleksii) [inline src/types/actionTypes.ts:24]: It's not readable, create a type for each action and then combine them with union type

File: src/types/apiErrorsType.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export type EmptyInputErrorType = 'REQUIRED';
+
+export const emptyInputError: EmptyInputErrorType = 'REQUIRED';
```
Comment (BudnikOleksii) [inline src/types/apiErrorsType.ts:3]: It's too much, if you set some string it gets only this string as type so you don't need to create another type for constant

**Review by BudnikOleksii — CHANGES_REQUESTED** [review]
Looks good 🚀 
I Left some comments to improve your solution, check them.
Also, a lot of tests failed:
image

**Comment by AndrMar1939** [comment]
> Looks good 🚀 I Left some comments to improve your solution, check them. Also, a lot of tests failed: image



> Looks good 🚀 I Left some comments to improve your solution, check them. Also, a lot of tests failed: image

Yes!!! I was trying to work with Bulma's additional styles, so the editing tests failed. Okay.

**Review by Moroz-Dmytro — CHANGES_REQUESTED** [review]
I see Oleksii has already reviewed it
Just some bug with the platform, I was not supposed to get this PR to review as it was not fixed yet
Need to reject this 🙂
Please, fix previous comments and I'll review it 😉

**Pull request closed without merging**